packages/fluxible-router/lib/handleHistory.js

```diff
@@ -210,9 +210,17 @@ function createComponent(Component, opts) {
         },
         _saveScrollPosition: function (e) {
             var historyState = (this._history.getState && this._history.getState()) || {};
+            var scrollX = window.scrollX || window.pageXOffset;
+            var scrollY = window.scrollY || window.pageYOffset;
+            // reduce unused replaceState
+            // also prevent IOS Safari reset scroll position to 0 with universal link bar showing
+            if (historyState.scroll && historyState.scroll.x === scrollX && historyState.scroll.y === scrollY) {
```

Comment:
I actually didn't aware that we have a wrapper for `history`, to me it makes sense to implement this logic in `_history.replaceState()` .